blitbuffer.c

```diff
@@ -416,6 +417,65 @@ void BB_fill_rect(BlitBuffer * restrict bb, unsigned int x, unsigned int y, unsi
     }
 }
 
+void BB_fill_rect_color(BlitBuffer * restrict bb, unsigned int x, unsigned int y, unsigned int w, unsigned int h, ColorRGB32 * restrict color) {
+    const int rotation = GET_BB_ROTATION(bb);
+    unsigned int rx, ry, rw, rh;
+    // Compute rotated rectangle coordinates & size
+    switch (rotation) {
+        case 0:
+                rx = x;
+                ry = y;
+                rw = w;
+                rh = h;
+                break;
+        case 1:
+                rx = bb->w - (y + h);
+                ry = x;
+                rw = h;
+                rh = w;
+                break;
+        case 2:
+                rx = bb->w - (x + w);
+                ry = bb->h - (y + h);
+                rw = w;
+                rh = h;
+                break;
+        case 3:
+                rx = y;
+                ry = bb->h - (x + w);
+                rw = h;
+                rh = w;
+                break;
+    }
+
+    // Handle any target pitch properly
+    const int bb_type = GET_BB_TYPE(bb);
+    switch (bb_type) {
+        case TYPE_BBRGB32:
+            // And here either, as we want to preserve the alpha byte
+            if (rx == 0 && rw == bb->w) {
+                // Single step for contiguous scanlines
+                const uint32_t src = (uint32_t) RGBA_To_RGB32(color->r, color->g, color->b, color->alpha);
```

Comment:
Unnecessary, ColorRGB32 is already an uint32_t in RGBA layout.

Reply:
Compiler isn't super happy about using the value as-is.

```
blitbuffer.c:460:28: warning: cast to smaller integer type 'uint32_t' (aka 'unsigned int') from 'ColorRGB32 *' (aka 'struct ColorRGB32 *') [-Wpointer-to-int-cast]
                    *p++ = (uint32_t) color;
```

Reply:
I think I want to do this `*p++ = *((uint32_t*)color);` (treat memory pointer as if it looking at uint32_t, then dereference that value)

Reply:
Yup, I'm fairly sure there are other instances of something like this (possibly sandwiched between GCC pragmas, because compilers don't like it).

Reply:
Ohkay, maybe not here, but in [FBInk](https://github.com/NiLuJe/FBInk/blob/af1a15b839462c3b19c611d4e3c6007dfc398da1/fbink.c#L235-L238) ;p.